packages/vitest/src/node/pool.ts

```diff
@@ -92,46 +112,63 @@ export function createPool(ctx: Vitest): ProcessPool {
 
     for (const spec of files) {
       const pool = getPoolName(spec)
-      if (!(pool in filesByPool))
-        throw new Error(`Unknown pool name "${pool}" for ${spec[1]}. Available pools: ${Object.keys(filesByPool).join(', ')}`)
+      filesByPool[pool] ??= []
       filesByPool[pool].push(spec)
     }
 
-    await Promise.all(Object.entries(filesByPool).map((entry) => {
+    const Sequencer = ctx.config.sequence.sequencer
+    const sequencer = new Sequencer(ctx)
+
+    async function sortSpecs(specs: WorkspaceSpec[]) {
+      if (ctx.config.shard)
+        specs = await sequencer.shard(specs)
+      return sequencer.sort(specs)
+    }
+
+    await Promise.all(Object.entries(filesByPool).map(async (entry) => {
```

Comment:
This issue seems to be in `main` branch too, but all pools are executed parallel here, right?

If user has `threads` pool and `vmThreads` pools set, and each are using default max-threads, the execution will spawn too many workers at the same time.

Reply:
Yes, this is how it works. That's why I wanted agnostic implementation of tinypool.

Reply:
Tinypool can run `worker_threads` and `child_process` parallel. 

https://github.com/tinylibs/tinypool/blob/999778ed8acccf18303760558f807f0e208c8b3b/test/runtime.test.ts#L151-L170

It's possible to push all `threads` tasks in the pool, then wait for queue to be empty (remaining tasks are running, nothing in queue), and then change the `runtime` and push more tasks in the queue to wait for previous ones to finish. Like here is done:

https://github.com/vitest-dev/vitest/blob/ab55637653c40c47d9f6ffe694f6c2335dc73260/packages/vitest/src/node/pools/threads.ts#L171-L174

Reply:
Ideally, I would like an API that we can expose to reuse the same Tinypool instance in different pools:

```ts
function createThreadsPool(ctx, { pool }) {
  function runTests(testFile) {
    const context = { config: ctx.config, files: [testFile] }
    await pool.run(context, { runtime: 'worker_threads', name: 'run' })
  }
}
```

Maybe instead of having separate `worker.js/child.js/vmWorker.js` files we have a single file where inside of a context we pass down information on how we want to run it? So, we don't need to change `filename` option in Tinypool.

Reply:
I like the idea. From Tinypool's side I think that kind of API should already be possible. 
There's also some need for refactoring in threads and forks pools to reduce code duplication. While doing that it might be good to look into this kind of API. But all this can be left out of this PR as it's not relevant to these changes.